File: yunj/level1/35_정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        double sqrt = Math.sqrt((double)n);
        if (sqrt - (long)sqrt > 0 ){
            return -1;
        }
        return (long)Math.pow(sqrt + 1, 2.0);
    }
}

File: sangjin/level1/35_정수_제곱근_판별.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    vector<long long> square(1, 0);

    for(long long i = 1; i * i <= 50000000000000; i++){
        square.push_back(i*i);
    }

    int num = (int)sqrt(n);

    if(n == square[num]){
         answer = square[num + 1];
    }else{
        answer = -1;
    }

    return answer;
}